File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/Separator.kt
package com.kinetixui.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

enum class KinetixSeparatorOrientation { Horizontal, Vertical }

/**
 * KinetixSeparator — mirrors `packages/ui/src/components/separator.tsx`
 * (`bg-border`, 1px thick, full-length on its own axis). Decorative by
 * default, same as the React source (Radix's `decorative` default).
 */
@Composable
fun KinetixSeparator(
    modifier: Modifier = Modifier,
    orientation: KinetixSeparatorOrientation = KinetixSeparatorOrientation.Horizontal,
) {
    val colors = KinetixColorScheme.current
    val axisModifier = when (orientation) {
        KinetixSeparatorOrientation.Horizontal -> Modifier.fillMaxWidth().height(1.dp)
        KinetixSeparatorOrientation.Vertical -> Modifier.fillMaxHeight().width(1.dp)
    }
    Box(modifier = modifier.then(axisModifier).background(colors.border))
}

File: packages/ui-swiftui/Sources/KinetixUI/Separator.swift
//
// Separator.swift — KinetixSeparator.
//
// Mirrors packages/ui/src/components/separator.tsx (`bg-border`, 1px
// thick, full-length on its own axis). Decorative — same as the Radix
// source's default.
//

import SwiftUI

public enum KinetixSeparatorOrientation {
    case horizontal, vertical
}

public struct KinetixSeparator: View {
    @Environment(\.kinetixColors) private var colors

    private let orientation: KinetixSeparatorOrientation

    public init(orientation: KinetixSeparatorOrientation = .horizontal) {
        self.orientation = orientation
    }

    public var body: some View {
        Rectangle()
            .fill(colors.border)
            .frame(width: orientation == .vertical ? 1 : nil)
            .frame(height: orientation == .horizontal ? 1 : nil)
            .frame(
                maxWidth: orientation == .horizontal ? .infinity : nil,
                maxHeight: orientation == .vertical ? .infinity : nil
            )
            .accessibilityHidden(true)
    }
}
